src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -0,0 +1,270 @@
+import { Cognite3DModel, Cognite3DViewer, THREE } from '@cognite/3d-viewer';
+import * as sdk from '@cognite/sdk';
+import React from 'react';
+import { CacheObject, Callback, MouseScreenPosition } from '../../interfaces';
+import {
+  createViewer as originalCreateViewer,
+  parseBoundingBox,
+  setCameraPosition,
+  ViewerEventTypes,
+} from '../../utils';
+
+let createViewer = originalCreateViewer;
+
+type ClickHandler = (position: MouseScreenPosition) => void;
+
+export interface Model3DViewerProps {
+  boundingBox: THREE.Box3;
+  cache: CacheObject;
+  useDefaultCameraPosition?: boolean;
+  modelId: number;
+  onError?: Callback;
+  onProgress?: Callback;
+  onComplete?: Callback;
+  onReady?: Callback;
+  onClick?: Callback;
+  onCameraChange?: Callback;
+  projectName: string;
+  revisionId: number;
```

Comment:
Could you order the props by required / not required?

Reply:
done